partiql-lang/src/jmh/kotlin/org/partiql/jmh/benchmarks/CompilerInterruptionBenchmark.kt

```diff
@@ -0,0 +1,307 @@
+        val exprValue = state.pipelineWithoutInterruption.compile(state.crossJoins)
+        blackhole.consume(exprValue)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun compileCrossJoinAggFuncWithInterruptible(state: MyState, blackhole: Blackhole) {
+        val exprValue = state.pipeline.compile(state.crossJoinsWithAggFunction)
+        blackhole.consume(exprValue)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun compileCrossJoinAggFuncWithoutInterruptible(state: MyState, blackhole: Blackhole) {
+        val exprValue = state.pipelineWithoutInterruption.compile(state.crossJoinsWithAggFunction)
+        blackhole.consume(exprValue)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun compileCrossJoinAggFuncGroupingWithInterruptible(state: MyState, blackhole: Blackhole) {
+        val exprValue = state.pipeline.compile(state.crossJoinsWithAggFunctionAndGroupBy)
+        blackhole.consume(exprValue)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun compileCrossJoinAggFuncGroupingWithoutInterruptible(state: MyState, blackhole: Blackhole) {
+        val exprValue = state.pipelineWithoutInterruption.compile(state.crossJoinsWithAggFunctionAndGroupBy)
+        blackhole.consume(exprValue)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun evalCrossJoinWithInterruptible(state: MyState, blackhole: Blackhole) {
+        val result = state.pipeline.compile(state.crossJoinsAst).evaluate(state.session) as PartiQLResult.Value
+        val value = result.value
+        blackhole.consume(value)
+    }
+
+    @Benchmark
+    @Fork(value = FORK_VALUE)
+    @Measurement(iterations = MEASUREMENT_ITERATION_VALUE, time = MEASUREMENT_TIME_VALUE)
+    @Warmup(iterations = WARMUP_ITERATION_VALUE, time = WARMUP_TIME_VALUE)
+    fun evalCrossJoinWithoutInterruptible(state: MyState, blackhole: Blackhole) {
+        val result = state.pipelineWithoutInterruption.compile(state.crossJoinsAst).evaluate(state.session) as PartiQLResult.Value
+        val value = result.value
+        blackhole.consume(value)
+    }
+
+//    TODO: See the note in the KDoc for the class.
```

Comment:
I understand that the eval test including aggregation is slow because we materialize all the data during aggregation. But I am not fully convinced that we need to comment those out through.

Reply:
They will run on every push to `main` according to our GH workflows.

Reply:
Point being the fact that they are slow does not interfere with the ability of them being a valid baseline for aggregation performance isn't it?

Reply:
Hmm wait. The measurement would be incorrect because those test includes materialization while others does not. If that is the case, then I would think commenting them out should be good. 

Maybe update the kdoc to reflect this?